### PR TITLE
Ispn 7080 NPE in CacheNotifierImpl by LIRS eviction listener

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-7080

### CODE REVIEW

File: core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java [hunk 1]
```diff
@@ -43,18 +49,28 @@ protected EmbeddedCacheManager createCacheManager() throws Exception {
...
       for (int i = 0; i < CACHE_SIZE; i++) {
-         cache.put("key-" + (i + 1), "value-" + (i + 1), 1, TimeUnit.MINUTES);
+         cache.put("key-" + (i + 1), "value-" + (i + 1));
+      }
+      assertEquals("cache size too big: " + cache.size(), CACHE_SIZE, cache.size());
+      int targetSize;
+      if (getEvictionStrategy() == EvictionStrategy.LIRS) {
+         // Eviction count will be Size + (Size * .95) rounded up (since the first elements will be in resident blocks
+         // so they won't cause evictions to occur
+         targetSize = (int) Math.ceil(CACHE_SIZE + (CACHE_SIZE * .05));
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java]: The variable name could be more specific, e.g. `expectedEvictions`.
Also, the comment says `Size + (Size * .95)` :)

Comment (wburns) [inline core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java]: Yeah outdated :P

**Comment by wburns** [comment]
Updated.

File: core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java [hunk 2]
```diff
@@ -28,22 +37,40 @@ protected BaseEvictionFunctionalTest() {
...
+      assertEquals("cache size too big: " + cache.size(), CACHE_SIZE, cache.size());
+      assertEquals("eviction events count should be same with case size: " + evictionListener.getEvictedEvents(),
+            CACHE_SIZE, evictionListener.getEvictedEvents().size());
+
+      for (int i = 0; i < CACHE_SIZE; i++) {
+         cache.put("key-" + (i + 1), "value-" + (i + 1));
       }
-      Thread.sleep(1000); // sleep long enough to allow the thread to wake-up
-      assert CACHE_SIZE >= cache.size() : "cache size too big: " + cache.size();
+      assertEquals("cache size too big: " + cache.size(), CACHE_SIZE, cache.size());
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java]: No need to put `cache.size()` in the message, TestNG will add it automatically.

File: commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/BoundedEquivalentConcurrentHashMapV8.java [hunk 3]
```diff
@@ -2737,27 +2725,33 @@ else if ((fh = f.hash) == MOVED)
...
-                                 if (oldVal == null) {
-                                    evictionPolicy.onEntryMiss(e, value);
-                                 } else {
-                                    evictionPolicy.onEntryHitWrite(e, value);
+                                 if (!skipListener) {
+                                    if (oldVal == null) {
+                                       evictionPolicy.onEntryMiss(e, value);
+                                    } else {
+                                       evictionPolicy.onEntryHitWrite(e, value);
+                                    }
```
Comment (danberindei) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/BoundedEquivalentConcurrentHashMapV8.java]: These are getting big enough now that IntelliJ is showing me a duplicate code warning :)

Comment (wburns) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/BoundedEquivalentConcurrentHashMapV8.java]: Well luckily they will be gone soon enough.

**Comment by wburns** [comment]
Fixed test message.

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Pull request closed without merging**